module-services/service-cellular/SMSParser.cpp

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include "SMSParser.hpp"
+
+#include <log/log.hpp>
+#include <ucs2/UCS2.hpp>
+
+#include <algorithm>
+#include <ctime>
+#include <map>
+#include <sstream>
+#include <unordered_map>
+#include <vector>
+
+namespace SMSParser
+{
+    namespace
+    {
+        static constexpr auto SMSReadMsgName        = "QCMGR";
+        static constexpr auto SingleMessageLength   = 5;
+        static constexpr auto MultipleMessageLength = 8;
+
+        std::unordered_map<std::uint16_t, std::map<std::uint8_t, std::string>> concatenatedMessage;
+        std::string message;
+        UTF8 number;
+    } // namespace
+
+    bool parse(const std::string *smsData)
+    {
+        bool parsed = false;
+
+        if (smsData[0].find(SMSReadMsgName) != std::string::npos) {
```

Comment:
What about inverting the condition and making early return here instead of enclosing all the code inside this `if`? Just a suggestion though, it's a matter of personal preference I suppose, feel free to close if you don't agree with me ;)

Reply:
I think that your suggestion is correct. It is done :)